src/main/java/io/anserini/index/IndexUtils.java

```diff
@@ -183,6 +194,98 @@ public void printDocumentVector(String docid) throws IOException, NotStoredExcep
     }
   }
 
+  public void dumpDocumentVectors(String reqDocidsPath, DocVectorWeight weight) throws IOException {
+    String outFileName = weight == null ? reqDocidsPath+".docvector.tar.gz" : reqDocidsPath+".docvector." + weight +".tar.gz";
+    LOG.info("Start Dump Document Vectors with weight " + weight);
+
+    InputStream in = getReadFileStream(reqDocidsPath);
+    BufferedReader bRdr = new BufferedReader(new InputStreamReader(in));
+    FileOutputStream fOut = new FileOutputStream(new File(outFileName));
+    BufferedOutputStream bOut = new BufferedOutputStream(fOut);
+    GzipCompressorOutputStream gzOut = new GzipCompressorOutputStream(bOut);
+    TarArchiveOutputStream tOut = new TarArchiveOutputStream(gzOut);
+
+    Map<Term, Integer> docFreqMap = new HashMap<>();
+    Map<String, String> docVectors;
```

Comment:
The definition can be moved to inside the loop?
We will introduce GC cycles anyway...

Reply:
The `docFreqMap` is a global variable for the entire loops. It stores the document frequency of every `Term` when it occurs and we query it from indexing, thus we can read it from memory next time we see this term. It'll be much slower if we query DF every time...

Reply:
Yes, what I meant is `docVectors`.

Reply:
Oh, sure, will do.